### DISCUSSION TITLE
How to use pipes in trascoding commands

### DISCUSSION THREAD

**Discussion by gandbg** [navidrome/navidrome#2337] [Q&A] [answered]
I'm trying to create a transcoding profile for HE-AACv2 with Navidrome running in Docker with the [fdkaac](https://github.com/nu774/fdkaac) command line wrapper for the Fraunhofer FDK AAC encoder.
Unfortunately, fdkaac can't decode audio files so I thought of using ffmpeg to do the decoding and some processing, then pass the uncompressed WAVE file to fdkaac for encoding.

Here's the command-line that I came up with:
```
ffmpeg -i %s -map 0:a -c:a pcm_s16le -map_metadata -1 -af volume=replaygain=track,aresample=resampler=soxr -ar 44100 -ac 2 -f wav - | fdkaac -f 2 -p 29 -b 16000 -o - -
```

However while this does work when run in a shell from the tweaked docker image (navidrome/navidrome:latest + fdkaac + Frounhoher FDK AAC library), it doesn't from Navidrome.

After some testing it seems that Navidrome doesn't run the commands though a shell but, instead runs the executable itself and so pipes wont work.

How could i run the same command though a shell that supports standard input/output?

Also no, i won't recompile ffmpeg. That's a too tedious and time-consuming process for me

**Comment by deluan** [comment, marked as answer]
Pipes should work. I'm pretty sure the commands are executed through a shell. I will confirm and test that, but in the meantime, one quick solution is to write a small shell script that calls the commands the way you want, and adding the shell script as the transcoding command.

**Reply by gandbg** [reply]
Sorry for the late reply and thanks for you suggestion. It works!
